java-checks/src/test/files/checks/TwoLocksWaitNotifyCheck.java

```diff
@@ -0,0 +1,51 @@
+
+class Wait {
```

Comment:
I tend to think that you lack a testcase with a local class  : 

```
class A {
   void foo() {
    synchronized (this.mon1) {
      synchronized (this.mon2) {
          class B {
                void bar() {
                     wait();
                }
          }
     }
  }
 }
}
```

And of course all the combination where that call can be hidden. 
Same idea : what about lambdas ? 
```
   void foo() {
    synchronized (this.mon1) {
      synchronized (this.mon2) {
              plop( a -> wait());
      }
    }
   }
```